sporkfish/endgame_tablebase.py

```diff
@@ -121,26 +184,33 @@ def query(self, board: Board) -> Optional[chess.Move]:
         :rtype: Optional[chess.Move]
         """
 
-        try:
-            if self._db:
-                if not self._should_query(board):
-                    return None
-
-                # We need to use a chess.Board() to be compatible with the endgame tablebase
-                # This is a small performace hit but is miniscule compared to searching
-                cboard = chess.Board()
-                cboard.set_fen(board.fen())
-                for move in cboard.legal_moves:
-                    cboard.push(move)
-                    # WDL Score: returns 2 if the side to move is winning, 0 if the position is a draw and -2 if the side to move is losing.
-                    # Returns 1 in case of a cursed win and -1 in case of a blessed loss.
-                    # Mate can be forced but the position can be drawn due to the fifty-move rule.
-                    # We check WDL score < 0 here, since we pushed our legal move and so its the opponents turn.
-                    # If they are losing, we are winning.
-                    if self._db.probe_wdl(cboard) < 0:
-                        return move
-                    cboard.pop()
-                return None
-            return None
-        except chess.syzygy.MissingTableError:
-            return None
+        if self._db and self._should_query(board):
+            # We need to use a chess.Board() to be compatible with the endgame tablebase
+            # This is a small performace hit but is miniscule compared to searching
+            cboard = chess.Board()
+            cboard.set_fen(board.fen())
+            best_move = None
+            best_dtz = -sys.maxsize
+            best_category = EndgameTablebase.DTZCategory.UNCONDITIONAL_LOSS
+            for move in board.legal_moves:
+                cboard.push(move)
+                # Refer to https://python-chess.readthedocs.io/en/latest/syzygy.html
+                # Invert the dtz as we are looking from the perspective of the opponent
+                dtz = -opp_dtz if (opp_dtz := self._db.get_dtz(cboard)) else None
+                cboard.pop()
+
+                if not dtz:
+                    continue
+
+                category = self._categorize_dtz(dtz)
+                if category > best_category:
```

Comment:
Sorry I am confused, if you swapped the DTZ score (i.e. instead of looking `opp_dtz`, we now look at own DTZ) - then surely the best category is UNCONDITIONAL_WIN? Surely it is easily to just work with `opp_dtz` directly?

I struggle to understand why if `category > best_category`, then the `best_dtz` is simply the `dtz` for the current `move`.

I think the logic here deserves an inline comment

Reply:
Actually this made me catch a bug, so ty :)